crates/bevy_asset/src/io/metadata.rs

```diff
@@ -42,10 +42,19 @@ impl TryFrom<std::fs::FileType> for FileType {
 /// This structure is returned from the [`AssetIo::get_metadata`] method.
 #[derive(Debug, Clone)]
 pub struct Metadata {
-    pub file_type: FileType,
+    file_type: FileType,
 }
 
 impl Metadata {
+    pub fn new(file_type: FileType) -> Self {
+        Self { file_type }
+    }
```

Comment:
I think this method is not needed? Should users be able to create directly a `Metadata`, or should it always be built from the `TryFrom<std::fs::Metadata>` impl?

Reply:
In my opinion, for `AssetIo` to be a true virtual filesystem it is important to not rely on `std::fs`, to support special cases of custom AssetIOs. For example in my project the assets are not directly on the filesystem but in archives. So I need to be able to create my own `Metadata`. To support such cases `Metadata` needs to be independent from `std::fs::Metadata`.